### ISSUE TITLE
add catergray for heimdall

### ISSUE THREAD

**Issue by xinmans** [linuxserver/Heimdall#870]
not tag

if you have menu more than 50， you need catergray to show, like deveop, pt download, nas, monitor etc.

could you support this feature?

**Comment by friki67** [comment]
I really like to see categories in Heimdall. If would make it perfect for me.

**Comment by Doublemyst** [comment]
Yes, this would be a great feature, similar to "homer" categories. This is the only thing which I really miss form homer.

**Comment by Chuk49** [comment]
I would really like to see that as well

**Comment by FudiHub** [comment]
+1 for this request.
Putting my apps into tags make me to click twice until I reach my app.
Just providing some sort of dividing/categories would be awesome.

**Comment by jniggemann** [comment]
+1 for this, that would be great.

**Comment by CapnDarkBeard** [comment]
+1 for me as well, This would be a great feature

**Comment by randomuser763543** [comment]
+1 great software, much appreciated on everything so far. Categories would be amazing on top

**Comment by xinmans** [comment]
now， i use [homepage ](https://github.com/benphelps/homepage) as a replace solution. it's very perfect

**Comment by jniggemann** [comment]
Thankfully, someone provided some code [on Reddit](https://www.reddit.com/r/selfhosted/comments/nzw76z/i_went_a_bit_overboard_customising_heimdall/) ([here's the pastebin](https://pastebin.com/5KBgQJ2W)). I tweaked it a bit and now I have categories using tags.

**Comment by mmol67** [comment]
Ok, I catch the idea. But there is something I really don't understand. Heimdall HAS categories. Just use tags to agrupate apps. 
![Screenshot_41](https://user-images.githubusercontent.com/60692795/232727140-95b66a2c-8c5c-4405-882b-5a6b76b51b2a.png)
![Screenshot_43](https://user-images.githubusercontent.com/60692795/232727166-258c065c-a7d6-4b91-9138-5c4000a690ed.png)
So, what are exactly categories for you?

**Comment by Doublemyst** [comment]
> Ok, I catch the idea. But there is something I really don't understand. Heimdall HAS categories. Just use tags to agrupate apps. So, what are exactly categories for you?

For me now those aren't really categories, but rather folders. 
In my case I don't want to hide elements into "folders" but rather have them all visible all the time, but neatly organized, so that all the media is in one "corner", the server stuff in another and so on.

Maybe you can check the Projects "Homer", or the "Homepage" as mentioned above.

**Comment by mmol67** [comment]
> For me now those aren't really categories, but rather folders. In my case I don't want to hide elements into "folders" but rather have them all visible all the time, but neatly organized, so that all the media is in one "corner", the server stuff in another and so on.
> 
> Maybe you can check the Projects "Homer", or the "Homepage" as mentioned above.

Yes. It is just another interface paradigm. I know these projects. I chose Heimdall because it looks great in my phone, it is easy to set up and has a lot of apps defined.

I like the idea of having all the things visible and access them in only one click through, so I can understand you.

**Comment by lazarus477** [comment]
**My problem**: If I tag an app it is visible in two places, on the main dashboard as well as in under the tagged listing. For me this is clutter, having the same item visible in two places. If I had the option to choose to only list the item when viewing the tagged listing then this would suffice.

What I am missing is the ability to have an app **pinned** in the **tag listing** while **unpinned** in the **main listing** of the **dashboard**. This would keep the app out of view until I click the tag. The basic concept of Main vs Sub items, like a menu system.

The other option which would also be a solution, as mentioned by @FudiHub, is to organize the apps visually into groups based on their tag.

Keeping tags separated from other dashboard items would further structure the layout. Tag items could be listed on a row of their own or on a side bar or menu.

**My goal is to:** reduce clutter and reduce the time it takes to find what the user is trying to find. Any form of structured organization would improve the user experience.

**Summary**: I think there is room for improvement in the form of refining the GUI. But this is what development is all about, refinement of things. Great app in all respects so consider this as constructive feedback for future improvements.

**Comment by mmol67** [comment]
> If I tag an app it is visible in two places

You only have to delete the "Home Dashboard" tag from the tag list in the application edit dialog:

![image](https://user-images.githubusercontent.com/60692795/235606376-27ff5e14-99a0-4382-a0f6-6ecba2100a44.png)

Regards.

**Comment by lazarus477** [comment]
Thank you @friki67, I had totally missed that tiny symbol, consequence of age combined with a small screen. This looks like it solves the problem for me.

**Comment by lazarus477** [comment]
Unless I am mistaken the **original issue**  has been resolved with the suggestion provided by @friki67.  Perhaps we can close it?

**Comment by FudiHub** [comment]
> Unless I am mistaken the **original issue** has been resolved with the suggestion provided by @friki67. Perhaps we can close it?

No the suggestion is not what meant here. The missing feature is to provide categories/groups of links on the dashboard as headlines. Similar than the tag feature but without having to click on it to show the links

**Comment by lazarus477** [comment]
**I agree** with the argument given by @FudiHub.

**Suggestion:**
Perhaps enough feedback has been gathered and this should all be documented as a feature request for a future release, such as 3.x?

**Comment by HelloMusician** [comment]
Yes, just like the file tree structure, clicking the 'Expand All' button allows us to see all applications grouped by categories on the current page, without having to click on a folder or category. Click this button again, and we can only see various categories or folders on the dashboard. Then the system will automatically record the page layout we last selected, and when we open Heimdall again, it will render the page according to our last selection. This can balance the two habits of users. Just for analysis and reference.

**Comment by HelloMusician** [comment]
> 是的，就像文件树结构一样，单击“全部展开”按钮可以让我们在当前页面上看到按类别分组的所有应用程序，而无需单击文件夹或类别。再次点击这个按钮，我们只能在仪表盘上看到各种类别或文件夹。然后系统会自动记录我们上次选择的页面布局，当我们再次打开Heimdall时，它会按照我们上次选择的方式渲染页面。这样可以平衡用户的两种习惯。仅供分析和参考。

But automatic deployment may increase the workload of UI adaptation, and I think the current functionality is already very good.
